v2/pgxcompat/uuid.go

```diff
@@ -0,0 +1,256 @@
+package pgxcompat
+
+import (
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgtype"
+
+	"github.com/SKF/go-utility/v2/uuid"
+)
+
+const uuidBinaryLength = 16
+const uuidStrShortLength = 32
+const uuidStrFullLength = 36
+
+var errUndefined = errors.New("cannot encode status undefined")
+var errBadStatus = errors.New("invalid status")
+
+type UUID struct {
+	UUID   uuid.UUID
+	Status pgtype.Status
+}
+
+type Getter interface {
+	Get() interface{}
+}
+
+func (u *UUID) Set(value interface{}) error { // nolint:gocyclo
+	if value == nil {
+		*u = UUID{Status: pgtype.Null}
+		return nil
+	}
+
+	if getterValue, ok := value.(Getter); ok {
+		value2 := getterValue.Get()
+		if value2 != getterValue {
+			return u.Set(value2)
+		}
+	}
+
+	switch value := value.(type) {
+	case uuid.UUID:
+		*u = UUID{UUID: value, Status: pgtype.Present}
+	case [16]byte:
+		uuidStr, _ := fromBinary(value[:]) // nolint:errcheck // we know input is 16 bytes
+		*u = UUID{UUID: uuid.UUID(uuidStr), Status: pgtype.Present}
+	case []byte:
+		if value == nil {
```

Comment:
Nilcheck no longer needed

Reply:
It's still needed, `([]byte)(nil)` which we have here is not the same as `interface{}(nil)` we check for in the beginning of the function. The same goes for `(*string)(nil)`, so I added that back.

But moving the first nil check to a case statement works, so I did that.

Reply:
How can value still be nil after it is converted into a []byte when we return in value == nil at line 32?

Does this mean that calling u.Set(nil) results in value == nil being true,
but calling 
mySlice []byte = nil
u.Set(mySlice) 
results in value == nil being false?

Reply:
Having read up on types nils, I'm starting to understand. I'll mark this as resolved.

Reply:
Yes: https://play.golang.org/p/McOzKQrvJCb